src/components/AlchemyRow.vue

```diff
@@ -0,0 +1,176 @@
+<template>
+  <td class="columnSpan">
+    <GameAsset class="centered"
+      :image="Assets.FromDir(`${props.group}${props.idx+1}`, 'alchemy')"
+      :height="64"/>
+  </td>
+  <td class="columnSpan">
+    <input class="columnSpan"
+      type="number" :min="0"
+      :value="alchemy.upgrades[group][props.idx]"
+      @change='customChange'/>
+  </td>
+  <td class="columnSpan">
+    <input class="columnSpan"
+      type="number" :min="alchemy.upgrades[group][props.idx]"
+      :value="alchemy.goals[group][props.idx]"
+      @change="handleAlchemyGoal($event, group, props.idx)"/>
+  </td>
+  <td class="centered columnSpan-2 effect"> 
+    {{effectChange()}}
+  </td>
+  <table class="filled">
+    <tr :style="{display: 'flex'}">
+      <td
+        :style="{width: '25%', whiteSpace: 'pre', marginRight: '5%'}" 
+        v-for="(thing, idx) in computeMaterials()"
+        :key="idx"
+      >
+        <th>{{thing[0]}}</th>
+        <tr :style="{display: 'inline-block', textAlign: 'right', width: '100%'}">
+          {{thing[1]}}
+          </tr>
+      </td>
+    </tr>
+  </table> 
+</template>
+
+
+<script lang="ts">
+import { computed, defineComponent, PropType,} from "vue";
+import { useState} from "~/State";
+import GameAsset from "~/components/GameAsset.vue";
+import { Assets, Growth } from "~/composables/Utilities";
+import { Color , Alch, Bubble} from "~/composables/Alchemy";
+import bubblesData from "~/data/bubbles.json"
+
+
+
+export default defineComponent({
+  name: "AlchemyRow",
+  components: {
+    GameAsset,
+  },
+  props: {
+    group: {
+      required: true,
+      type: String as PropType<Color>,
+    },
+    discount: {
+      required: true,
+      type: Array as PropType<number[]>,
+    },
+    idx: {
+      required: true, 
+      type: Number,
+    }
+  },
+  emits: ["custom-change"],
+  methods: {
+    customChange (event: { target: { value: string; id: number; }; }) {
+      console.log("Handle leveling bubble input")
+      console.log("Only emit when id == 14 or (id == 6 and group == yellow)")
+      this.alchemy.upgrades[this.props.group][this.props.idx] = parseInt(event.target.value);
+      if (this.props.idx == 14 || (this.props.idx == 6 && this.props.group == "Yellow")) {
+        this.$emit("custom-change", event.target.value);
+        console.log(`Emitted [${15-event.target.id}] ${event.target.value}`);
+      }
+    }
+  },
+  setup(props) {
+    const state = useState();
+
+    const alchemy = computed({
+      get: () => state.value.alchemy,
+      set: (value) => (state.value.alchemy = value),
+    });
+   
+    // TODO:  This could maybe move to the customChange method
+    const handleAlchemyGoal = (ev: Event, color: Color, i: number) => {
+      let target = <HTMLInputElement>ev.target;
+      let val = target ? target.value : "0";
+      alchemy.value.goals[color][i] = parseInt(val);
+    };
```

Comment:
When the PR that introduces Ant design lands, we can use their components for strict numeric inputs. This is just unfortunate boilerplate we need for now.